### PR TITLE
Benefit Summaries Async

### PR DESCRIPTION
**Pull request by MattEberhart-MSFTE**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json [hunk 1]
```diff
@@ -314,6 +314,636 @@
...
+            "description": "Async Benefit Utilization Summary report to be created.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BenefitUtilizationSummariesRequest"
+            }
+          }
+        ],
+        "responses": {
+          "202": {
```
Comment (pilor) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:355]: >202 [](http://example.com/codeflow?start=11&length=3)

this needs a "200" response as well containing the final result schema.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json [hunk 2]
```diff
@@ -314,6 +314,636 @@
...
+          "url": "https://docs.microsoft.com/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "GenerateUtilizationSummariesReportByBillingAccount": {
+            "$ref": "./examples/BenefitUtilizationSummaries/Async/GenerateBenefitUtilizationSummariesReportByBillingAccount.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
```
Comment (pilor) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:335]: >location [](http://example.com/codeflow?start=30&length=8)

if any of these async POST APIs are exposing what could be considered personal data or secrets and you need to release these within the next month or two please implement step 4 here:

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#operation-result-rules](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#operation-result-rules)

ARM will be introducing a way to enforce that centrally in the near future but its not ready yet

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json (hunk 1 above)
Comment (pilor) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:355]: Its how the SDK knows what to deserialize the result into after following the polling URIs  (the manual GET operations you have below are separate and optional)

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json [hunk 3]
```diff
@@ -863,6 +1493,120 @@
...
+          "type": "string",
+          "format": "date-time"
+        }
+      }
+    },
+    "BenefitUtilizationSummaryReportSchema": {
+      "description": "The CSV file from the reportUrl and secondaryReportUrl blob link will consist of the following columns of benefit utilization data. UtilizedPercentage will be 0 for savings plans reports and non data bricks reservations. Utilization percentages will be 0 for data bricks reservations.",
+      "type": "string",
+      "enum": [
+        "BenefitKind",
```
Comment (sri-) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:1551]: >BenefitKind [](http://example.com/codeflow?start=9&length=11)

Do we call this Kind of BenefitKind in the code?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json [hunk 4]
```diff
@@ -863,6 +1493,120 @@
...
+          "description": "The start date of the summaries data that will be served in the report.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "endDate": {
+          "description": "The end date of the summaries data that will be served in the report.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "benefitKind": {
```
Comment (sri-) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:1605]: >benefitKind [](http://example.com/codeflow?start=9&length=11)

Should this be benefitType ?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json [hunk 5]
```diff
@@ -931,6 +1691,14 @@
           }
         ]
       }
+    },
+    "benefitsAsyncOperationIdParameter": {
```
Comment (sri-) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:1695]: nit: place in alphabetical order

Comment (sri-) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:1695]: should this be benefitAsyncOperationIdParameter instead?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json (hunk 1 above)
Comment (MattEberhart-MSFTE) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:355]: This API won't return 200. The operation results API is the polling API which will return 200 with final result schema

Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:355]: Even if your API is not returning 200 response, it is needed for Swagger/SDK completeness, as Chris explained in previous comment. Currently, there is no way for SDK to know what is the schema of the response.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json [hunk 6]
```diff
@@ -314,6 +314,636 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/generateBenefitUtilizationSummariesReport": {
+      "post": {
+        "tags": [
+          "BenefitUtilizationSummariesAsync"
+        ],
+        "operationId": "GenerateBenefitUtilizationSummariesReportAsync_BillingAccountScope",
```
Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json:323]: >GenerateBenefitUtilizationSummariesReportAsync_BillingAccountScope [](http://example.com/codeflow?start=24&length=66)

See the Linter Warnings: PostOperationIdContainsUrlVerb.

You need to correct the operationId format. For example, here it should be "BillingAccountScope_GenerateBenefitUtilizationSummariesReport". Similarly for other Post Action APIs also.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**